src/mmap.cpp

```diff
@@ -104,18 +104,32 @@ MemoryMap::MemoryMap()
 	bootRomFile = nullptr;
 	romFile = nullptr;
 
-	audio = new APU();
-
 	mbcMode = 0x0;
 }
 
+// Push to audio write queue
+void MemoryMap::pushAudioWriteQueue(Word address, Byte value)
+{
+	audioRegs temp = { address, value };
+	MemoryMap::audioWriteQueue.push(temp);
+}
+
+// remove the first element from queue
+audioRegs MemoryMap::popAudioWriteQueue()
+{
+	audioRegs t = audioWriteQueue.front();
+	audioWriteQueue.pop();
+	return t;
+};
+
 // Write to memory
 // TODO: Make emulation memory secure
-bool MemoryMap::writeMemory(Word address, Byte value)
+bool MemoryMap::writeMemory(Word address, Byte value, bool audioWrite)
```

Comment:
Why is this bool needed?

Reply:
I needed that for write back after executing APU::writeByte. But if I were to write back, the write was again pushed to the queue creating an infinite cycle and hence to distinguish write from write back that bool is necessary.